decidim-vocdoni/decidim-module-vocdoni
# Add "Results published" step in admin

This PR adds the "Vote period ended" and "Results published" steps in admin panel.

It fetches the results using the Vocdoni SDK's `fetchElection` method, and saves them to the Decidim database.

We also drop some unnecessary steps (tally_started and tally_ended), as both of these steps are done transparently to us by the Vocdoni API.

Mind that even though the step is called "Results published" we're actually not publishing them yet, only saving them. 

## Screenshots

![Selection_511](https://user-images.githubusercontent.com/717367/219115631-63e60f04-15ae-4d67-8bc8-3a48fad2b936.png)
![Selection_512](https://user-images.githubusercontent.com/717367/219115647-c8bda1ae-7d05-47a5-aac4-5508b4b5e88b.png)

## Testing

1. Set up an Election
1. Vote on it
1. Wait until it ends
2. Run the bin/rails `decidim_vocdoni:change_election_status` task so it changes the step
3. Go to the steps
4. See that you have the number of votes by questions/answers that corresponds with whatever you vote
5. Click on the "Publish results"
6. Check that they're saved on the next page. You can also check them out in the `answer.votes` column
